x-pack/metricbeat/module/azure/app_insights/data.go

```diff
@@ -19,45 +21,80 @@ func EventsMapping(metricValues insights.ListMetricsResultsItem, applicationId s
 	if metricValues.Value == nil {
 		return events
 	}
+	groupedAddProp := make(map[string][]insights.MetricsResultInfo)
 	for _, item := range *metricValues.Value {
 		if item.Body != nil && item.Body.Value != nil {
 			if item.Body.Value.AdditionalProperties != nil {
-				events = append(events, createEvent(*item.Body.Value, insights.MetricsSegmentInfo{}, applicationId))
+				groupedAddProp[fmt.Sprintf("%sTO%s", item.Body.Value.Start, item.Body.Value.End)] =
+					append(groupedAddProp[fmt.Sprintf("%sTO%s", item.Body.Value.Start, item.Body.Value.End)], *item.Body.Value)
```

Comment:
Consider calculating the key in a single place to avoid mistakes.
```suggestion
				key := fmt.Sprintf("%sTO%s", item.Body.Value.Start, item.Body.Value.End)
				groupedAddProp[key] = append(groupedAddProp[key], *item.Body.Value)
```